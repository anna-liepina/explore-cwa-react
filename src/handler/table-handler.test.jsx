import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import TableHandler from './table-handler';

describe('<TreeHandler/>', () => {
    const props = {
        form: { config: [] }
    };

    const optionalProps = {
        'data-cy': 'optProps.data-cy',
        className: 'optProps.className',
        title: 'optProps.title',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { asFragment, debug } = render(<TableHandler {...props} />);

            expect(asFragment()).toMatchSnapshot();
        });

        it('with optional/required props', () => {
            const { asFragment } = render(<TableHandler {...props} {...optionalProps} />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
