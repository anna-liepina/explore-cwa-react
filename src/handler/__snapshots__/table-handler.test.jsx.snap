// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeHandler/> render with default/required props 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <div
      class="form_controls"
    />
  </form>
  <table
    class="table"
  >
    <thead>
      <tr
        class="table-row table-row-header"
      />
    </thead>
    <tbody
      class="table-content"
    />
  </table>
</DocumentFragment>
`;

exports[`<TreeHandler/> render with optional/required props 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <div
      class="form_controls"
    />
  </form>
  <table
    class="table"
  >
    <thead>
      <tr
        class="table-row table-row-header"
      />
    </thead>
    <tbody
      class="table-content"
    />
  </table>
</DocumentFragment>
`;
